GitCommands/Git/ExecutableExtensions.cs

```diff
@@ -158,11 +190,23 @@ string ComposeOutput()
         [MustUseReturnValue("Callers should verify that " + nameof(RunBatchCommand) + " returned true")]
         public static bool RunBatchCommand(
             this IExecutable executable,
-            ArgumentString[] batchArguments,
+            ICollection<BatchArgumentItem> batchArguments,
+            Action<BatchProgressEventArgs> action = null,
             byte[] input = null,
             bool createWindow = false)
         {
-            return batchArguments.Aggregate(true, (result, arguments) => executable.RunCommand(arguments, input, createWindow) && result);
+            int total = batchArguments.Sum(item => item.Count);
+            var result = true;
+
+            foreach (var item in batchArguments)
+            {
+                result = executable.RunCommand(item.Argument, input, createWindow) && result;
```

Comment:
This looks odd. Why are we using conditional AND?
```suggestion
                result &= executable.RunCommand(item.Argument, input, createWindow);
```